angular-bind-html-compile.js

```diff
@@ -6,15 +6,23 @@
     module.directive('bindHtmlCompile', ['$compile', function ($compile) {
         return {
             restrict: 'A',
+            scope: {
+                bindHtmlCompile: '=',
+                bindHtmlScope: '=?'
+            },
             link: function (scope, element, attrs) {
-                scope.$watch(function () {
-                    return scope.$eval(attrs.bindHtmlCompile);
-                }, function (value) {
+                scope.$watch('bindHtmlCompile', function (value) {
                     // Incase value is a TrustedValueHolderType, sometimes it
                     // needs to be explicitly called into a string in order to
                     // get the HTML string.
                     element.html(value && value.toString());
-                    $compile(element.contents())(scope);
+                    if (scope.bindHtmlScope) {
+                        // If a scope was provided, use it
+                        $compile(element.contents())(scope.bindHtmlScope);
+                    } else {
+                        // Otherwise use parent scope
+                        $compile(element.contents())(scope.$parent);
```

Comment:
This could be made simpler:
```js
// Use the provided scope, otherwise use the parent scope.
var compileScope = scope.bindHtmlScope || scope.$parent;
$compile(element.contents())(compileScope);
```